wordpress-mobile/WordPress-Android
# Stats Revamp v2: Follower Types card

Fixes #16204 

This adds follower types card to the total followers detail screen. 

#### Do not merge before these PRs:
- [x] #16545
- [x] wordpress-mobile/WordPress-FluxC-Android#2400

Design|PR|
|-|-|
|<img src="https://user-images.githubusercontent.com/2471769/168444775-5964bbb6-35fd-40ec-b78e-a7ff99a8389a.png" width=320>|<img src="https://user-images.githubusercontent.com/2471769/169365934-0d6f17ee-b7bb-48bd-9fe1-6255e8961421.png" width=320>|


### Notes
- 0.3% is rounding to 0% and "2 (0%)" looks wrong at first glance. **Need feedback.**
- `contentDescription` for "WordPress.com 627 (99%)" is "WordPress.com 627". "99%" is not being mentioned. **Need feedback.**
- The card doesn't have a donut graph, for now, it'll be added in a separate PR.
- The percent sign isn't localized. An issue is open for it #16491 

To test:
1. Go to App Settings (Tap on Avatar at the top right-hand corner on My Site -> Find App Settings).
2. Select Debug Settings.
3. Find StatsRevampV2FeatureConfig under Features in development enable it and restart app.
4. Launch the app.
5. Go to stats either using quick links or menu.
6. Ensure that it opens in Insights tab otherwise switch to Insights tab.
7. Notice a new Total Followers card is shown in Insights tab with Line Chart (scroll down if necessary).
8. Tap on "VIEW MORE" on the top right-hand corner of the card.
9. Ensure it opens the detail screen as shown above (scroll down if necessary) with Follower Types card.
10. Check and compare follower counts with values in WordPress.com stats page.

## Regression Notes
1. Potential unintended areas of impact
Current insight types in Add New Card screen could be impacted.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested.

3. What automated tests I added (or what prevented me from doing so)
`FollowerTypesUseCaseTest`

PR submission checklist:
